From d51b3db2211e806891384798c0c028502bb2b50b Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 14 Nov 2014 04:37:10 -0500
Subject: [PATCH 1/4] Don't hide GtkBuilder errors

This makes it unnecessarily hard to track down invalid
ui files.

https://bugzilla.gnome.org/show_bug.cgi?id=740138
---
 libseahorse/seahorse-widget.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/libseahorse/seahorse-widget.c b/libseahorse/seahorse-widget.c
index 963dfd2..49284e4 100644
--- a/libseahorse/seahorse-widget.c
+++ b/libseahorse/seahorse-widget.c
@@ -259,6 +259,7 @@ object_set_property (GObject *object, guint prop_id, const GValue *value, GParam
     SeahorseWidget *swidget;
     GtkWidget *w;
     char *path;
+    GError *error = NULL;
     
     swidget = SEAHORSE_WIDGET (object);
     
@@ -270,7 +271,12 @@ object_set_property (GObject *object, guint prop_id, const GValue *value, GParam
         path = g_strdup_printf ("%sseahorse-%s.xml",
                                 SEAHORSE_UIDIR, swidget->name);
         swidget->gtkbuilder = gtk_builder_new ();
-        gtk_builder_add_from_file (swidget->gtkbuilder, path, NULL);
+        gtk_builder_add_from_file (swidget->gtkbuilder, path, &error);
+        if (error)
+          {
+            g_warning ("Error parsing %s: %s\n", path, error->message);
+            g_error_free (error);
+          }
         g_free (path);
         g_return_if_fail (swidget->gtkbuilder != NULL);
         
-- 
2.1.0

